### PR TITLE
Cleanup config model, part 1

### PR DESCRIPTION
**Pull request by hmusum**
Remove unused code, optimize imports, fix method signatures.
No functional changes.

### CODE REVIEW

**Review by baldersheim — APPROVED** [review]

File: config-model/src/main/java/com/yahoo/searchdefinition/derived/IndexInfo.java [hunk 1]
```diff
@@ -11,7 +11,6 @@
 import com.yahoo.search.config.IndexInfoConfig;
 
 import java.util.*;
```
Comment (baldersheim) [inline config-model/src/main/java/com/yahoo/searchdefinition/derived/IndexInfo.java]: Perhaps remove .*

File: config-model/src/main/java/com/yahoo/vespa/model/container/search/QueryProfiles.java [hunk 2]
```diff
@@ -14,7 +14,6 @@
 import java.util.*;
```
Comment (baldersheim) [inline config-model/src/main/java/com/yahoo/vespa/model/container/search/QueryProfiles.java]: Remove .*

File: config-model/src/test/java/com/yahoo/searchdefinition/FieldOfTypeDocumentTestCase.java [hunk 3]
```diff
@@ -4,7 +4,6 @@
 import com.yahoo.document.*;
```
Comment (baldersheim) [inline config-model/src/test/java/com/yahoo/searchdefinition/FieldOfTypeDocumentTestCase.java]: Remove .*

File: config-model/src/test/java/com/yahoo/vespa/model/builder/xml/dom/DomConfigPayloadBuilderTest.java [hunk 4]
```diff
@@ -18,7 +18,6 @@
 
 import javax.xml.parsers.ParserConfigurationException;
 import java.io.*;
```
Comment (baldersheim) [inline config-model/src/test/java/com/yahoo/vespa/model/builder/xml/dom/DomConfigPayloadBuilderTest.java]: .*

File: config-model/src/test/java/com/yahoo/vespa/model/content/StorageContentTest.java [hunk 5]
```diff
@@ -4,16 +4,13 @@
...
 import com.yahoo.vespa.model.VespaModel;
 import com.yahoo.vespa.model.routing.DocumentProtocol;
 import com.yahoo.vespa.model.routing.Routing;
 import com.yahoo.vespa.model.test.utils.ApplicationPackageUtils;
 import com.yahoo.vespa.model.test.utils.VespaModelCreatorWithMockPkg;
 import org.junit.Test;
-import org.xml.sax.SAXException;
 
-import java.io.IOException;
 import java.util.*;
```
Comment (baldersheim) [inline config-model/src/test/java/com/yahoo/vespa/model/content/StorageContentTest.java]: .*

File: config-model/src/main/java/com/yahoo/vespa/model/container/search/QueryProfiles.java (hunk 2 above)
Comment (hmusum) [reply config-model/src/main/java/com/yahoo/vespa/model/container/search/QueryProfiles.java]: Fixed

File: config-model/src/test/java/com/yahoo/searchdefinition/FieldOfTypeDocumentTestCase.java (hunk 3 above)
Comment (hmusum) [reply config-model/src/test/java/com/yahoo/searchdefinition/FieldOfTypeDocumentTestCase.java]: Fixed

File: config-model/src/test/java/com/yahoo/vespa/model/builder/xml/dom/DomConfigPayloadBuilderTest.java (hunk 4 above)
Comment (hmusum) [reply config-model/src/test/java/com/yahoo/vespa/model/builder/xml/dom/DomConfigPayloadBuilderTest.java]: Fixed

File: config-model/src/test/java/com/yahoo/vespa/model/content/StorageContentTest.java (hunk 5 above)
Comment (hmusum) [reply config-model/src/test/java/com/yahoo/vespa/model/content/StorageContentTest.java]: Fixed

File: config-model/src/main/java/com/yahoo/searchdefinition/derived/IndexInfo.java (hunk 1 above)
Comment (hmusum) [reply config-model/src/main/java/com/yahoo/searchdefinition/derived/IndexInfo.java]: Fixed

**Comment by baldersheim** [comment]
Merge at will when done

**Pull request merged**